pkg/controllers/restore-controller/handler.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2020 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restorecontroller
+
+import (
+	"fmt"
+	apis "github.com/openebs/api/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/cstor-operators/pkg/controllers/common"
+	"github.com/openebs/cstor-operators/pkg/volumereplica"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+	"os"
+	"reflect"
+)
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the CStorReplicaUpdated resource
+// with the current status of the resource.
+func (c *RestoreController) syncHandler(key string, operation common.QueueOperation) error {
+	klog.Infof("Sync handler called for key:%s with op:%s", key, operation)
+	rst, err := c.getCStorRestoreResource(key)
+	if err != nil {
+		return err
+	}
+	if rst == nil {
+		return fmt.Errorf("can not retrieve CStorRestore %q", key)
+	}
+
+	if IsDoneStatus(rst) || IsFailedStatus(rst) {
+		return nil
+	}
+
+	status, err := c.rstEventHandler(operation, rst)
+	if status == "" {
+		return nil
+	}
+
+	if err != nil {
+		klog.Errorf(err.Error())
+		rst.Status = apis.RSTCStorStatusFailed
+	} else {
+		rst.Status = apis.CStorRestoreStatus(status)
+	}
+
+	nrst, err := c.clientset.OpenebsV1alpha1().CStorRestores(rst.Namespace).Get(rst.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	nrst.Status = rst.Status
+
+	_, err = c.clientset.OpenebsV1alpha1().CStorRestores(nrst.Namespace).Update(nrst)
+	if err != nil {
+		return err
+	}
+
+	klog.Infof("Completed operation:%v for restore:%v, status:%v", operation, nrst.Name, nrst.Status)
+	return nil
+}
+
+// eventHandler will execute a function according to a given operation
+func (c *RestoreController) rstEventHandler(operation common.QueueOperation, rst *apis.CStorRestore) (string, error) {
+	switch operation {
+	case common.QOpAdd:
```

Comment:
Qn: Why we followed the old way of writing controllers?

Reply:
Did not scope to refactor now reason being it can then become Risky for RC1. 
Will continue to refactor as part of the next PRs.